File: 75-sort-colors/sort-colors.js

var sortColors = function(nums) {
    let start = 0
    let mid = 0
    let end = nums.length - 1

    while (mid <= end) {
        if (nums[mid] == 0) {
            [nums[start], nums[mid]] = [nums[mid], nums[start]]
            start++
            mid++
        }
        else if (nums[mid] == 1) mid++

        else if (nums[mid] == 2) {
            [nums[mid], nums[end]] = [nums[end], nums[mid]]
            end--
        }
    }    
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        mid = 0
        j = len(nums) - 1

        while (mid <= j):
            if (nums[mid] == 0):
                nums[i], nums[mid] = nums[mid], nums[i]
                i += 1
                mid += 1
            elif (nums[mid] == 1):
                mid += 1
            elif (nums[mid] == 2):
                nums[mid], nums[j] = nums[j], nums[mid]
                j -=1

    
        